src/common/procedure/src/local.rs

```diff
@@ -496,13 +507,30 @@ impl LocalManager {
 
                 if let Err(e) = self.submit_root(
                     *procedure_id,
+                    init_state_loader(message),
```

Comment:
We could add a `set_error` method to `InitProcedureState` to attach the error message. Then we don't need to define add a generic fn to this method.